test/Hpack/Syntax/DependencySpec.hs

```diff
@@ -73,36 +76,20 @@ spec = do
               github: yesodweb/wai
               ref: master
               subdir: warp
-          |] `shouldDecodeTo_` Dependencies [("warp", SourceDependency source)]
+          |] `shouldDecodeTo_` Dependencies [("warp", defaultInfo { dependencyVersion = SourceDependency source })]
 
         it "accepts local dependencies" $ do
           let source = Local "../hpack"
           [yaml|
             - name: hpack
               path: ../hpack
-          |] `shouldDecodeTo_` Dependencies [("hpack", SourceDependency source)]
-
-        context "when ref is missing" $ do
-          it "produces accurate error messages" $ do
-            [yaml|
-              - name: hpack
-                git: sol/hpack
-                ef: master
-            |] `shouldDecodeTo` left "Error while parsing $[0] - key \"ref\" not present"
```

Comment:
I'm eager to keep this error messages as accurate as possible.

Reply:
The current PR version (pre-rebase) is going to emit an 'unused field' warning for these. (Or it should. I'll re-add tests.) Is that good enough, or does it have to be a hard error?

Reply:
It's a hard error with current `master`.  We just keep the exiting tests in place.